rust/processor/src/models/ans_models/ans_lookup.rs

```diff
@@ -0,0 +1,776 @@
+                    transaction_timestamp: txn_timestamp,
+                },
+            );
+        }
+
+        // If the name record doesn't exist yet, create it and insert it to current_ans_lookups.
+        // This is possible if the primary name record wsc happens before the name record wsc.
+        new_current_ans_lookups.insert(
+            (
+                primary_name_record.get_domain().clone(),
+                primary_name_record.get_subdomain().clone(),
+            ),
+            Self {
+                domain: primary_name_record.get_domain().clone(),
+                subdomain: primary_name_record.get_subdomain().clone(),
+                registered_address: Some(String::from(key)),
+                last_transaction_version: txn_version,
+                // Default to zero
+                expiration_timestamp: chrono::NaiveDateTime::from_timestamp_opt(0, 0)
+                    .unwrap_or_default(),
+                token_name: primary_name_record.get_token_name(),
+                is_primary: true,
+                is_deleted: false,
+            },
+        );
+        new_ans_lookups.insert(
+            (
+                txn_version,
+                primary_name_record.get_domain().clone(),
+                primary_name_record.get_subdomain().clone(),
+            ),
+            AnsLookup {
+                transaction_version: txn_version,
+                domain: primary_name_record.get_domain(),
+                subdomain: primary_name_record.get_subdomain(),
+                registered_address: Some(String::from(key)),
+                // Default to zero
+                expiration_timestamp: chrono::NaiveDateTime::from_timestamp_opt(0, 0)
+                    .unwrap_or_default(),
+                token_name: primary_name_record.get_token_name(),
+                is_primary: true,
+                is_deleted: false,
+                transaction_timestamp: txn_timestamp,
+            },
+        );
+
+        Ok((new_current_ans_lookups, new_ans_lookups))
+    }
+
+    pub fn from_primary_name_record_table_delete_table_item_v1(
+        delete_table_item: &DeleteTableItem,
+        txn_version: i64,
+        txn_timestamp: chrono::NaiveDateTime,
+        address_to_primary_name_record: &AddressToPrimaryNameRecord,
+        conn: &mut PgPoolConnection,
+    ) -> anyhow::Result<(
+        HashMap<CurrentAnsLookupPK, Self>,
+        HashMap<AnsLookupPK, AnsLookup>,
+    )> {
+        let mut new_current_ans_lookups = HashMap::new();
+        let mut new_ans_lookups = HashMap::new();
+
+        let table_item_data = delete_table_item.data.as_ref().unwrap();
+        let key_type = table_item_data.key_type.as_str();
+        let key = serde_json::from_str(&table_item_data.key).unwrap_or_else(|e| {
+            panic!(
+                "version {} failed! failed to parse type {}, data {:?}. Error: {:?}",
+                txn_version, key_type, table_item_data.key, e
+            )
```

Comment:
same comment as above.

Reply:
Oop this got left over from when we had a combined loop for both. Will remove.